joomla/joomla-cms
# Improving menu item workflow in Joomla[GSoC 16_menu_for_article_project]

### Summary of Changes

In Joomla 1.0 it was possible to create a menu link to an article from the article edit view in the administrator back-end. Joomla is missing this feature since 1.5 and it has a long time since then. Content-Createmenulink plugin recreates the old memories of creating menu item directly from the component view.
### Testing Instructions
1. Install the patch/PR.
2. Go to Extensions > Plugins.
3. Enable Content-Createmenulink Plugin if it is disabled.
4. Click on Content-Createmenulink plugin.
5. Parameter dialogue view will appear.
   ![firstpr1](https://cloud.githubusercontent.com/assets/12000068/17907191/53158230-6998-11e6-9907-bfaede689cba.png)
6. Enter component+view in the form [com_content.article], just like in the given image below.
   Type two component view to enable component in both component view.
   [Note: Remember to press enter before typing second component+view name]

![firstpr2](https://cloud.githubusercontent.com/assets/12000068/17907253/b1230d48-6998-11e6-9e8a-71d0a310c8d7.png)
1. Navigate to content > article >create new article.
2. A new tab named as "Menu link options" will appear.
   ![firstpr3](https://cloud.githubusercontent.com/assets/12000068/17907351/2eb066d4-6999-11e6-84a7-55d1c1fea31f.png)
3. Fill all the details.
4. Check fornt end.
5. Also try to make a new menu item for contact or other 3rd party component using this PR.
   ![firstpr4](https://cloud.githubusercontent.com/assets/12000068/17907406/82d224b4-6999-11e6-95ce-89b8a3003eaf.png)
### Expected outcome

New menu item for the article/contact/3rd party component will get created in the frontend.
